src/adapter/src/catalog/builtin_table_updates/notice.rs

```diff
@@ -17,37 +17,9 @@ use mz_transform::notice::{
     RawOptimizerNotice,
 };
 
-use crate::catalog::{BuiltinTableUpdate, Catalog, CatalogState};
+use crate::catalog::{BuiltinTableUpdate, Catalog};
 
 impl Catalog {
-    /// Transform the [`DataflowMetainfo`] by rendering an [`OptimizerNotice`]
-    /// for each [`RawOptimizerNotice`].
-    ///
-    /// Delegates to [`CatalogState::render_notices`].
-    pub fn render_notices(
-        &self,
-        df_meta: DataflowMetainfo<RawOptimizerNotice>,
-        notice_ids: Vec<GlobalId>,
-        item_id: Option<GlobalId>,
-    ) -> DataflowMetainfo<Arc<OptimizerNotice>> {
-        self.state.render_notices(df_meta, notice_ids, item_id)
-    }
-
-    /// Pack a [`BuiltinTableUpdate`] with the given `diff` for each
-    /// [`OptimizerNotice`] in `notices` into `updates`.
-    ///
-    /// Delegates to [`CatalogState::pack_optimizer_notices`].
-    pub fn pack_optimizer_notices<'a>(
-        &self,
-        updates: &mut Vec<BuiltinTableUpdate>,
-        notices: impl Iterator<Item = &'a Arc<OptimizerNotice>>,
-        diff: Diff,
-    ) {
-        self.state.pack_optimizer_notices(updates, notices, diff);
-    }
-}
-
-impl CatalogState {
```

Comment:
The reason why these were implemented under `CatalogState` and not `Catalog` (with delegating methods) is in order to mirror the other `pack_~` methods which seem to have been recently moved to `CatalogState` by @benesch. I'm not sure whether hard-coding the dependency to `Catalog` (because of Catalog::for_system_session()` has other implications.

Reply:
Sure thing--moved that back to `CatalogState` for consistency with similar methods.